great_tables/_formats.py

```diff
@@ -285,66 +286,129 @@ def fmt_number(
     sep_mark = _get_locale_sep_mark(default=sep_mark, use_seps=use_seps, locale=locale)
     dec_mark = _get_locale_dec_mark(default=dec_mark, locale=locale)
 
-    # Generate a function that will operate on single `x` values in the table body
-    def fmt_number_fn(
-        x: float | None,
-        decimals: int = decimals,
-        n_sigfig: int | None = n_sigfig,
-        drop_trailing_zeros: bool = drop_trailing_zeros,
-        drop_trailing_dec_mark: bool = drop_trailing_dec_mark,
-        use_seps: bool = use_seps,
-        scale_by: float = scale_by,
-        compact: bool = compact,
-        sep_mark: str = sep_mark,
-        dec_mark: str = dec_mark,
-        force_sign: bool = force_sign,
-    ):
-        if is_na(self._tbl_data, x):
-            return x
-
-        # Scale `x` value by a defined `scale_by` value
-        x = x * scale_by
-
-        # Determine whether the value is positive
-        is_negative = _has_negative_value(value=x)
-
-        if compact:
-            x_formatted = _format_number_compactly(
-                value=x,
+    # TODO: fix type errors for members of `FormatFns`
+    return fmt(
+        self,
+        fns=FormatFns(
+            html=partial(
```

Comment:
While it's hard to see in the diff, the `FormatFns()` call will always use three named parameters: `html`, `latex`, and `default`. Each of these will have a `partial` statement so that variables are injected as default values. The args of `partial()` are always of this pattern/order:

- the function `fmt_*_context()`
- `data=self`
- other args for the `fmt_*_context()` function
- `context=`, where the context string is hard-coded, matching the named parameter in `FormatFns()`

The `fmt_*_context()` has an `x` argument which is untouched by `partial()`. When the function is later invoked, `x` will be the list of values (e.g., from a column).